src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -3572,34 +3572,55 @@ public S(int arg)
         }
 
         [Theory, CombinatorialData]
-        public void InStruct_Cycle(bool firstIsSemi, bool secondIsSemi)
+        public void InStruct_Cycle(bool firstIsSemi, bool secondIsSemi, bool bindS1First)
```

Comment:
I wanted to test all the combinations, especially when both are semi-auto

Reply:
I don't think the complexity of the current test is worth the effort. This test is trying to verify that, if a semi-auto property is bound second, there will be an extra bind of the property body. That detail is getting lost amid the muddle of "wait, what does `S1` have in this part of the theory?". If you want to test both of them having a semi-auto or neither having one, it would be better to make a separate test, imo.

Reply:
Should be fixed now.